strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitChangesDetector.java

```diff
@@ -75,7 +93,7 @@ private File findRepoRoot(File currentDir) {
             .map(diffEntry -> {
                 try {
                     final File sourceFile = new File(repoRoot, diffEntry.getNewPath());
-                    return new ClassNameExtractor().extractFullyQualifiedName(sourceFile);
+                    return ClassNameExtractor.extractFullyQualifiedName(sourceFile);
```

Comment:
Well my idea was just to not instantiate the object in every call, so making an attribute of the class might be an option too. But also static for me it would has sense, but for me is more related to performance of creation and destruction of the object.